### ISSUE TITLE
`dotnet test` string-backed, flag-style parameters require arguments incorrectly

### ISSUE THREAD

**Issue by baronfel** [dotnet/sdk#23250]
### Describe the bug

There are several forwarded parameters in the `dotnet test` command parser that are of type `Option<string>` but act like boolean flags. When set these forward along multiple string arguments to a child process.  An example of this would be the `--blame-crash-collect-always` flag. This flag, and others like it, error in SDK `6.0.100` when called with no argument parameter, which isn't in line with their intended use.  These flags should be changed to not require an argument in order to work.

### To Reproduce

In a project on the 6.0.100 SDK, run `dotnet test --blame-crash-collect-always`

### Exceptions (if any)

> Required argument missing for option: --blame-crash-collect-always

### Further technical details
.NET SDK 6.0.100 release

There are two separate problems that I think we'll run into:
* The current implementation doesn't specify arity for these string options, and so System.CommandLine is defaulting to an arity of ExactlyOne for these string-typed parameters. This is what's at the core of requiring the argument. This can be fixed in a few different ways:
  * Explicitly set arity in the constructors of these specific options to ZeroOrOne
  * Implicitly set arity by changing the type of the options to `bool` or `bool?` (because bools and any nullable value type are set to ZeroOrOne arity by the library)
* However, If you change the arity to ZeroOrOne, the interaction of the `ForwardAsMany` lambdas changes so that those values are _always_ passed along. To change this the lambda need to be changed to check the value of the parsed argument being passed in, and if the value is `false` (for `bool` options), or `null` (for nullable options) then return `Enumerable.Empty<string>()` instead of the filled list.

**Comment by baronfel** [comment]
Implementation Note:

I think a ParseArgument delegate like the following would be the most reusable way of ensuring the flag's default boolean value of `true` is provided:

```cs
        private static readonly ParseArgument<bool> FlagStyleArgument =
         (argumentResult) =>
            {
                if (argumentResult.Parent is not OptionResult or)
                {
                    throw new NotSupportedException("The method should be only used with option.");
                }

                if (argumentResult.Tokens.Count == 0)
                {
                    return true; // if the argument is implicit, then use 'true' as the default value.
                }
                else if (argumentResult.Tokens.Count == 1)
                {
                    if (bool.TryParse(argumentResult.Tokens[0].Value, out bool value))
                    {
                        return value;
                    }
                    argumentResult.ErrorMessage = $"Could not parse '{argumentResult.Tokens[0].Value}' as expected type 'bool'.";
                    return false;
                }
                else
                {
                    argumentResult.ErrorMessage = $"Using more than 1 argument is not allowed for '{or.Token.Value}', used: {argumentResult.Tokens.Count}.";
                    return false;
                }
            };
```

**Comment by BartoszKlonowski** [comment]
I would like to work on this.
@baronfel @marcpopMSFT Please assign me to this item.

**Comment by Tunduk** [comment]
@baronfel Hi! 
What about `Zero` value for arity? I think it is more suitable for boolean flag-style options. 
There is no word about arguments for this kind options in docs. I am not sure that someone uses `--blame-crash-collect-always true`. It will be a breaking change of course, but will someone notice it?

**Comment by baronfel** [comment]
pinging @nohwnd for confirmation, but I think he recently did a pass through the arities of many of the `dotnet test` options to make sure they were accurate. He had convincing reasons why the booleans should stay - specifically scripting, where the true/false argument might be dynamically determined.

**Comment by Tunduk** [comment]
Yeah, I forget about script scenario. Good point. Maybe we should add this info in the docs then

**Comment by nohwnd** [comment]
You are right @baronfel I wrote full example here: https://github.com/dotnet/sdk/pull/26963#issuecomment-1210700289 

Specifically for usage with PowerShell (which is common in our repos), it is the most natural thing to do is `--no-restore:$NoRestore`, which is broken with zero arity, and you would need to do ugly inline if.
